config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/file/JsonFileUtils.java

```diff
@@ -84,4 +86,13 @@ public BundleDefinedEntities readBundleMetadataFile(final File metaDataFile) {
             throw new JsonFileUtilsException("Error reading the bundle metadata file " + metaDataFile.toString(), e);
         }
     }
+
+    public Object readBundleMetadataFile(final String bundleFolderPath, final String fileName) {
+        final File metaDataFile = new File(bundleFolderPath, fileName + METADATA_FILE_NAME_SUFFIX);
+        if (metaDataFile.exists()) {
+            final MapType type = jsonTools.getObjectMapper(YAML_EXTENSION).getTypeFactory().constructMapType(LinkedHashMap.class, String.class, Object.class);
```

Comment:
Please make sure the order of the fields in the metadata is maintained after replacing it.